doc/developer/arrangements.md

```diff
@@ -274,3 +268,15 @@ But, we would also need to create similar dataflow graphs for each of `In2`, `In
 In each case they only require arrangements of the input, but they may be by different keys.
 
 If a `Join` is implemented by a Delta Join pattern, it will create zero additional arrangements.
+
+## Caveats: Demand Analysis
+
+> Obsolete as of v0.9.4. We now delete fields that are not required
+> instead of blanking them out. Plans prior to v0.9.4 will show something like
+> `| | demand = (#6, #8, #12, #15, #22, #23, #27)` for the `Join` and `FlatMap`
+> operators, listing which field will be blanked out.
+
+When users present queries and views to us, we can determine that some fields are not required.
+We blank out any field that is not required.
+This can reduce the number of distinct `data` in an arrangement, which will
+reduce the size of the arrangement.
```

Comment:
Nit, but: as of now we still do the demand analysis, and still (afaict) propagate that information to e.g. sources in the form of `LinearOperator`.

Reply:
Sorry, this should highlight the section above, the "Obsolete as of v0.9.4." part. I think it's not yet obsolete, until we remove `demand.rs`.